testutil/log.go

```diff
@@ -0,0 +1,29 @@
+package testutil
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func Log(tag string, message ...interface{}) {
+	if len(message) > 0 {
+		log(tag, message[0])
+	} else {
+		log(tag, nil)
+	}
+}
+
+func log(tag string, a interface{}) {
+	if a == nil {
+		fmt.Println(tag)
+		return
+	}
+
+	// a!=nil
+	array := reflect.ValueOf(a)
+	if array.Kind() == reflect.Array || array.Kind() == reflect.Slice {
```

Comment:
More avoidable complexity.  What about the spew stuff you wired up?  I really hate seeing `reflect` used so liberally like this.  An exception I make is `reflect.DeepEqual`, when appropriate.